pkg/typesystem/typesystem_create_test.go

```diff
@@ -0,0 +1,426 @@
+package typesystem
+
+import (
+	"testing"
+
+	openfgav1 "github.com/openfga/api/proto/openfga/v1"
+	"github.com/stretchr/testify/require"
+
+	"github.com/openfga/openfga/internal/condition"
+	"github.com/openfga/openfga/pkg/testutils"
+)
+
+// usersetsEquals tests that the two *openfgav1.Usersets
+// values provided are equal. This is a deep equality check.
+func usersetsEquals(t *testing.T, a, b *openfgav1.Usersets) {
+	if a == nil || b == nil {
+		require.Equal(t, a, b)
+		return
+	}
+	childA := a.GetChild()
+	childB := b.GetChild()
+	require.Equal(t, len(childA), len(childB))
+	for i := 0; i < len(childA); i++ {
+		rewriteEquals(t, childA[i], childB[i])
+	}
+}
+
+// tupleToUsersetEquals tests that the two *openfgav1.TupleToUserset
+// values provided are equal. This is a deep equality check.
+func tupleToUsersetEquals(t *testing.T, a, b *openfgav1.TupleToUserset) {
+	if a == nil || b == nil {
+		require.Equal(t, a, b)
+		return
+	}
+	cuA := a.GetComputedUserset()
+	cuB := b.GetComputedUserset()
+
+	if cuA == nil || cuB == nil {
+		require.Equal(t, cuA, cuB)
+	}
+	objectRelationEquals(t, cuA, cuB)
+
+	tsA := a.GetTupleset()
+	tsB := b.GetTupleset()
+	if tsA == nil || tsB == nil {
+		require.Equal(t, tsA, tsB)
+	}
+	objectRelationEquals(t, tsA, tsB)
+}
+
+// objectRelationEquals tests that the two *openfgav1.ObjectRelation
+// values provided are equal. This is a deep equality check.
+func objectRelationEquals(t *testing.T, a, b *openfgav1.ObjectRelation) {
+	if a == nil || b == nil {
+		require.Equal(t, a, b)
+		return
+	}
+	require.Equal(t, a.GetRelation(), b.GetRelation())
+	require.Equal(t, a.GetObject(), b.GetObject())
```

Comment:
**(suggestion):** When using the `require` package from `github.com/stretchr/testify`, if the assertion fails, the test will stop and we won't continue. This however seems like a good case for introducing the `assert` package that has the exact same usage, that won't stop the test run but make multiple assertions together instead and report all the issues.

```suggestion
	assert.Equal(t, a.GetRelation(), b.GetRelation())
	assert.Equal(t, a.GetObject(), b.GetObject())
```

Reply:
You make a good point. I'll make this change.